src/CodeStyle/Core/Analyzers/AnalyzerConfigOptions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis
+{
+    internal abstract class OptionSet
```

Comment:
@agocke This is a copy of `AnalyzerConfigOptions` which is used as the primary options mechanism in this pull request. I named it `OptionSet` for now to maximize code sharing until the transition is ready, but it should at least demonstrate the sufficiency of the upcoming APIs.